### DISCUSSION TITLE
How to reference components from other components?

### DISCUSSION THREAD

**Discussion by TwilightMage** [bevyengine/bevy#13309] [Q&A] [answered]
So, this is my use case.
I have one `Terrain` component on a scene. `Terrain` does management of chunks (`Chunk`) spawned across the world. At some moment I may need to get chunk by some position (if it exist), so I want to store `Chunk` references at dictionary. What type should I use?

**Comment by SkiFire13** [comment, marked as answer]
You can use a `Entity` to "reference" (not in the sense of holding an actual Rust reference though) another entity. To get a `&Chunk` out of it you'll have to take a `Query<&Chunk>` and call `get` on it passing the `Entity`.

**Reply by TwilightMage** [reply]
Yeah, that might work. I just add from myself that you need to pass `&mut World` around to construct query on a fly and also to fetch the component, but that's pretty much similar to UE world context.
